ext/sqlite3/sqlite3.c

```diff
@@ -1945,11 +1949,25 @@ PHP_METHOD(SQLite3Result, fetchArray)
 				RETURN_FALSE;
 			}
 
+			if (result_obj->column_count == -1) {
+				result_obj->column_count = sqlite3_column_count(result_obj->stmt_obj->stmt);
+			}
+
+			zend_long n_cols = result_obj->column_count;
+
+			/* Cache column names to speed up repeated fetchArray calls. */
+			if (mode & PHP_SQLITE3_ASSOC && !result_obj->column_names) {
+				result_obj->column_names = emalloc(n_cols * sizeof(zend_string*));
+
+				for (int i = 0; i < n_cols; i++) {
+					const char *column = sqlite3_column_name(result_obj->stmt_obj->stmt, i);
+					result_obj->column_names[i] = zend_string_init(column, strlen(column), 0);
```

Comment:
You might want to check whether using zend_string_init_existing_interned() would be beneficial in practice (you'd have to measure that together with code using the data).